pkg/secrets/provide_conjur_secrets_test.go

```diff
@@ -14,11 +14,31 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+const (
+	providerDelayTime = 500
+)
+var providerCount = 0
```

Comment:
I think the reason that we're seeing some flaky test cases below (the ones that are commented out) is that we're using one common `providerCount` gobal variable for all test cases. It makes testing more complex when you're testing slow provider functionality because you have to wait for some grace period at the end of each test case for the previous test case's slow provider to finish.

Generally, we want each test case to have its own state. So this should be implemented as a structure that has a `provide()` method, and instantiate one of those for each test case.

To see what I mean, check out this mocking:
https://github.com/cyberark/secrets-provider-for-k8s/commit/d61c96c091f171be0aa04be5c75a6407a3e5f092#diff-c0314624500148cb0ccac2409ef0d4dc9a8cddf7603e8879bae326cc2d96d7d5R17-R52

This mocking works, except that I haven't reconciled it with the `eventualProvider()` function, which depends upon those base objects.